includes/RestApi/SettingsController.php

```diff
@@ -206,13 +208,30 @@ public function get_current_settings() {
 			// update database
 			\add_option( Options::get_option_name( 'wpseo_social', false ), $social_data );
 		}
-		$twitter_handle = $this->validate_twitter_id( $social_data['twitter_site'] );
 		// add the full url for twitter cause only the handle is saved in the database
-		if ( ! empty( $social_data['twitter_site'] ) &&
-			( false !== $twitter_handle ) ) {
+		$twitter_handle = $this->validate_twitter_id( $social_data['twitter_site'] );
+		if ( ( ! empty( $social_data['twitter_site'] ) ) && ( false !== $twitter_handle ) ) {
 			$social_data['twitter_site'] = 'https://www.twitter.com/' . $twitter_handle;
 		}
 
+		// handle other social urls for onboarding
+		foreach ( $social_data['other_social_urls'] as $index => $social_url ) {
+			// remove the indexed url from array
+			unset( $social_data['other_social_urls'][ $index ] );
+
+			switch ( $social_url ) {
+				case ( -1 < stripos( $social_url, 'yelp.com' ) ):
```

Comment:
Let's use that regex we discussed about?

Reply:
regex check added